Fix Groups#index

noriyotcp opened the pull request:
もともと`groups/index.html.slim`では`current_user`の`groups`しか表示されていなかった
にもかかわらず、`Groups#index`では`Group.all`で全グループを取得していた

`current_user.groups`のみを取得するようにした

現在の`/groups` の画面

![2018-08-31 16 02 16](https://user-images.githubusercontent.com/5820754/44897666-6be86900-ad37-11e8-9107-fd77827bef08.png)

itkrt2y commented on spec/controllers/groups_controller_spec.rb:
```diff
@@ -14,7 +14,23 @@
+
     it { is_expected.to render_template :index }
+
+    context 'when user has 2 memberships' do
```
このテストいらないです

itkrt2y commented:
余談ですが、 `/groups` はバグで表示できなかったものを僕が表示できるように直したもので、現状だと本番では開けない、使われてるのか疑わしいページだったりします

noriyotcp committed:
Fix Groups#index

もともとgroups/index.html.slimではcurrent_userのgroupsしか表示されていなかった
にもかかわらず、Groups#indexではGroup.allで全グループを取得していた

current_user.groupsのみを取得するようにした

noriyotcp commented:
テスト削除しました。確かに`/groups` への導線は見た感じなさそうですね

itkrt2y commented:
いや、導線はあるけどバグってて絶対に開けない
